OrbitGl/CrashManager.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CrashManager.h"
+
+#include <thread>
+
+#include "OrbitBase/Logging.h"
+#include "services.grpc.pb.h"
+
+namespace {
+
+class CrashManagerImpl final : public CrashManager {
+ public:
+  explicit CrashManagerImpl(std::shared_ptr<grpc::Channel> channel);
+
+  void CrashOrbitService(GetCrashRequest_CrashType crash_type) override;
+  void Shutdown() override;
+
+ private:
+  void WorkerFunction(GetCrashRequest_CrashType crash_type);
+
+  std::unique_ptr<CrashService::Stub> crash_service_;
+  std::thread crash_thread_;
+};
+
+CrashManagerImpl::CrashManagerImpl(std::shared_ptr<grpc::Channel> channel)
+    : crash_service_(CrashService::NewStub(channel)) {}
+
+void CrashManagerImpl::Shutdown() {
+  if (crash_thread_.joinable()) {
+    crash_thread_.join();
+  }
+}
+
+void CrashManagerImpl::WorkerFunction(GetCrashRequest_CrashType crash_type) {
+  GetCrashRequest request;
+  request.set_crash_type(crash_type);
+
+  GetCrashResponse response;
+  grpc::ClientContext context;
+
+  grpc::Status status =
+      crash_service_->CrashOrbitService(&context, request, &response);
+  if (!status.ok()) {
+    ERROR("Grpc call failed: %s", status.error_message());
+  }
+}
+
+void CrashManagerImpl::CrashOrbitService(GetCrashRequest_CrashType crash_type) {
```

Comment:
Let the caller do thread management here - there a thread_pool for this. Just make a gprc call here. You can then remove Shutdown since this will no longer be needed.

Reply:
It makes code easier, thanks! Updated pull request